daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala

```diff
@@ -53,12 +57,56 @@ object Speedy {
       var localContracts: Map[V.ContractId, (Ref.TypeConName, SValue)],
       // global contract discriminators, that are discriminators from contract created in previous transactions
       var globalDiscriminators: Set[crypto.Hash],
+      /* Used when enableLightweightStepTracing is true */
+      var steps: Int,
```

Comment:
We could wrapped this variable and the following ones in a new object (call something like statics) to limit the extension this endless list of parameter. 
You will pay the small cost of indirection only when instrumentation is running.